fpdf/table.py

```diff
@@ -268,23 +280,124 @@ def _render_table_cell(
                 if style
                 else FontFace(fill_color=self._cell_fill_color)
             )
-        with self._fpdf.use_font_face(style):
-            page_break, cell_height = self._fpdf.multi_cell(
-                w=col_width,
-                h=row_height,
-                txt=cell.text,
-                max_line_height=self._line_height,
-                border=self.get_cell_border(i, j),
-                align=text_align,
-                new_x="RIGHT",
-                new_y="TOP",
-                fill=fill,
-                markdown=self._markdown,
-                output=MethodReturnValue.PAGE_BREAK | MethodReturnValue.HEIGHT,
-                wrapmode=self._wrapmode,
-                **kwargs,
+
+        if cell.padding:
+            from fpdf.fpdf import get_padding_tuple
+
+            padding = get_padding_tuple(cell.padding)
+        else:
+            padding = self._padding
+
+        if cell.v_align:
+            v_align = cell.v_align
+        else:
+            v_align = self._v_align
+
+        # place cursor (required for images after images)
+        cell_widhts = [self._get_col_width(i, jj) for jj in range(j)]
+        cell_x = sum(cell_widhts)
+        self._fpdf.set_x(self._fpdf.l_margin + cell_x + self._gutter_width * j)
+
+        # render cell border and background
+
+        # if cell_height is defined, that means that we already know the size at which the cell will be rendered
+        # so we can draw the borders now
+        #
+        # If cell_height is None then we're still in the phase of calculating the height of the cell meaning that
+        # we do not need to render and borders yet.
+        if cell_height is not None:
+            with self._fpdf.use_font_face(style):
```

Comment:
Why do you need to the font there?
If it's to set the "fill color", it would clearer to just:
```python
prev_fill_color = self._fpdf.fill_color
if style.fill_color:
    self._fpdf.set_fill_color(style.fill_color)
... do stuff
if style.fill_color:  # we restore the initial fill_color:
    self._fpdf.set_fill_color(prev_fill_color)
```

Reply:
It is just about the fill-color indeed. But I prefer the use of a context-manager over setting and restoring a value.

Reply:
Alright, but the `use_font_face()` context-manager is about setting temporary font-related settings.

Hence, it is a bit misleading there

Reply:
Combine this comment with the _draw_box / rect() comment and incorporate the fill color in _draw_box. This removes the context manager.